### ISSUE TITLE
Why adminRefs?

### ISSUE THREAD

**Issue by jeffreydavis83** [chaoming/fireact#4]
Hi Chaoming Li, thanks for this amazing starter template. I am new to web dev so this has been much of help.

I just wanted to ask that why do we need "adminRefs" in our database? Like we already have the "admin" array to check if user has the permissions.

**Comment by chaoming** [comment]
The `adminRefs` and `userRefs` are for reading the user documents under the `users` collection.

**Comment by jeffreydavis83** [comment]
Makes sense. Thanks!

**Comment by jeffreydavis83** [comment]
Sorry I am gonna ask another noob question,

What if we don’t use the `adminRefs` and `userRefs` and instead fetch from the database using id only? I am just curious how  `adminRefs` and `userRefs` are benefiting us?

**Comment by chaoming** [comment]
No worries.

There are two approaches to not use `adminRefs` and `userRefs`.

1) After reading the user IDs from `access` or `admins` array, loop through the `users` collection one by one based on the list of IDs. I think this is definitely not a very efficient way to retrieve data because each ID will need an API call to the `users` collection to retrieve the user document.

2) After reading the user IDs from `access` or `admins` array, execute a query to retrieve all the user documents from the `users` collection that match the IDs in the array. However, I can't find examples to query a collection based on a list of IDs in the document https://firebase.google.com/docs/firestore/query-data/queries, so I designed the system to use array of reference to resolve it. This is possible based on https://github.com/googleapis/nodejs-firestore/issues/990, but I am not sure about the performance compared to the current method.

Cheers,

**Comment by jeffreydavis83** [comment]
Thanks man! I just have this one doubt,

The cloud function `getAccountUsers` includes:

```
 let getUsers = [];
            accountRef.data().accessRefs.forEach(user => {
                getUsers.push(getDoc('users/'+user.id));
            });
            return Promise.all(getUsers);
```

Aren’t you executing `getDoc` for each user? Wouldn’t each user then will require an API call in this case?

**Comment by chaoming** [comment]
True, looks like it's OK to remove `adminRefs` and `accessRefs` and simply to use `admins` and `access` to simplify the data structure. Thanks for the question.

**Comment by chaoming** [comment]
This issue is resolved in commit dd1286c71192523d237adb5d00d38657e3a549e1

**Comment by jeffreydavis83** [comment]
Thanks!
